swig/include/python/docs/ogr_fielddomain_docs.i

```diff
@@ -0,0 +1,236 @@
+%feature("docstring") OGRFieldDomainShadow "
+
+Python proxy of an :cpp:class:`OGRFieldDomain`.
+
+Created using one of:
+
+- :py:func:`CreateCodedFieldDomain`
+- :py:func:`CreateGlobFieldDomain`
+- :py:func:`CreateRangeFieldDomain`
+";
+
+%extend OGRFieldDomainShadow {
+
+%feature("docstring")  GetDescription "
+
+Get the description of the field domain.
+
+See :cpp:func:`OGRFieldDomain::GetDescription`.
+
+.. versionadded:: 3.3
+
+Returns
+--------
+str
+    the field domain description (might be empty string).
+";
+
+%feature("docstring")  GetName "
+
+Get the name of the field domain.
+
+See :cpp:func:`OGRFieldDomain::GetName`.
+
+.. versionadded:: 3.3
+
+Returns
+--------
+str
+    the field domain name.
+";
+
+%feature("docstring")  GetDomainType "
+
+Get the type of the field domain.
+
+See :cpp:func:`OGRFieldDomain::GetDomainType`.
+
+.. versionadded:: 3.3
+
+Returns
+--------
+int
+    the type of the field domain.
+
+Examples
+--------
+>>> d = ogr.CreateCodedFieldDomain('my_code', None, ogr.OFTInteger, ogr.OFSTNone, { 1 : 'owned', 2 : 'leased' })
+>>> d.GetDomainType() == ogr.OFDT_CODED
+True
+
+";
+
+%feature("docstring")  GetEnumeration "
+
+Get the enumeration as (code, value) pairs.
```

Comment:
while true for the C API, it is perhaps a bit confusing to speak about "pairs" in the context of the Python API, given that we use a Python dict? feel free to disregard as I feel I'm a bit nitpicking :-)

Reply:
"Get the enumeration as mapping of codes to values." ?

Reply:
> Get the enumeration as mapping of codes to values." ?

if that sounds better for you, I'm happy with that too.